modules/core/src/directx.cpp

```diff
@@ -1392,10 +1392,150 @@ void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst
         __convertFromD3D11Texture2DKHR(pD3D11Texture2D, dst);
     }
     else {
-        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initilized without DirectX interoperability");
+        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initialized without DirectX interoperability");
+    }
+#endif
+}
+
+#ifdef HAVE_OPENCL
+void convertFromD3D11Texture2DtoCLMem(ID3D11Texture2D* pD3D11Texture2D, void* clMem, void* clContext, void* clDeviceID)
+{
+    CV_UNUSED(pD3D11Texture2D); CV_UNUSED(clMem);
+#if !defined(HAVE_DIRECTX)
+    NO_DIRECTX_SUPPORT_ERROR;
+#elif !defined(HAVE_OPENCL)
+    NO_OPENCL_SUPPORT_ERROR;
+#else
+    ID3D11Device *pD3D11Device = nullptr;
```

Comment:
clDeviceID and etc - need to be checked on !nullptr at least

Reply:
Added